eq-author-api/schema/resolvers/pages/calculatedSummaryPage.js

```diff
@@ -60,29 +63,24 @@ Resolvers.CalculatedSummaryPage = {
   availablePipingAnswers: ({ id }, args, ctx) =>
     getPreviousAnswersForPage(ctx.questionnaire, id),
   availablePipingMetadata: (page, args, ctx) => ctx.questionnaire.metadata,
-  validationErrorInfo: ({ id }, args, ctx) => {
-    const calculatedSummaryErrors = ctx.validationErrorInfo.filter(
-      ({ pageId }) => id === pageId
-    );
-
-    return ({
-      id,
-      errors: calculatedSummaryErrors,
-      totalCount: calculatedSummaryErrors.length,
-    });
-  },
+  validationErrorInfo: ({ id }, args, ctx) =>
+    returnValidationErrors(ctx, id, ({ pageId }) => id === pageId),
 };
 
 Resolvers.Mutation = {
   createCalculatedSummaryPage: createMutation(
-    (root, { input: { position, sectionId } }, ctx) => {
-      const section = find(ctx.questionnaire.sections, {
-        id: sectionId,
-      });
+    (root, { input: { position, sectionId, folderId } }, ctx) => {
       const page = createCalculatedSummary({ sectionId });
-      const insertionPosition =
-        typeof position === "number" ? position : section.pages.length;
-      section.pages.splice(insertionPosition, 0, page);
+      if (folderId) {
+        const folder = getFolderById(ctx, folderId);
+        folder.pages.splice(getPosition(position, folder.pages), 0, page);
+      } else {
+        const folders = getFoldersBySectionId(ctx, sectionId);
+        const folder = createFolder({
+          pages: [page],
+        });
+        folders.push(folder);
+      }
```

Comment:
Because confirmation pages are always tied to a specific question maybe it would make more sense to add the confirmation page to the question's folder (for the case when folderId isn't supplied in the query)? Might be preferable to putting it in a new empty folder because if the question is skipped then the confirmation presumably would be too?

Reply:
The highlighted section of code is for a calculated summary page though. Do you mean those instead?

Reply:
Oops, my bad! Thought it was for confirmation pages, haha. This is fine for calc sum. Ignore me ;-)